File: commons-text/src/main/java/pl/wrzasq/commons/text/formatter/MarkdownFormatter.java
/*
 * This file is part of the pl.wrzasq.commons.
 *
 * @license http://mit-license.org/ The MIT license
 * @copyright 2016, 2019 © by Rafał Wrzeszcz - Wrzasq.pl.
 */

package pl.wrzasq.commons.text.formatter;

import org.pegdown.Extensions;
import org.pegdown.PegDownProcessor;

/**
 * Markdown format handler.
 */
public class MarkdownFormatter implements FormatterInterface
{
    /**
     * Markdown processor with all available extras.
     */
    private PegDownProcessor markdownProcessor = new PegDownProcessor(Extensions.ALL);

    /**
     * {@inheritDoc}
     */
    @Override
    public String transform(String text)
    {
        return this.markdownProcessor.markdownToHtml(text);
    }
}

File: commons-text/src/main/kotlin/pl/wrzasq/commons/text/formatter/MarkdownFormatter.kt
/*
 * This file is part of the pl.wrzasq.commons.
 *
 * @license http://mit-license.org/ The MIT license
 * @copyright 2016, 2019 © by Rafał Wrzeszcz - Wrzasq.pl.
 */

package pl.wrzasq.commons.text.formatter

import org.intellij.markdown.flavours.MarkdownFlavourDescriptor
import org.intellij.markdown.flavours.gfm.GFMFlavourDescriptor
import org.intellij.markdown.html.HtmlGenerator
import org.intellij.markdown.parser.MarkdownParser

/**
 * Markdown format handler.
 *
 * @param flavour Markdown flavour handler.
 */
class MarkdownFormatter(
    private val flavour: MarkdownFlavourDescriptor = GFMFlavourDescriptor()
) : FormatterInterface {
    override fun transform(text: String): String {
        val parsedTree = MarkdownParser(flavour).buildMarkdownTreeFromString(text)
        return HtmlGenerator(text, parsedTree, flavour).generateHtml().removeSurrounding("<body>", "</body>")
    }
}
